### PR TITLE
[Feature] Added SASL_SSL Plain authentication for Kafka trigger scalar to work with Event Hubs

### PR DESCRIPTION
**Pull request by rasavant-ms**
- Added SASL_SSL Plain authentication for Kafka Trigger Scalar.
- Removed extra newline characters being read from secrets file

This enables connecting Event Hub w/Kafka head as a scalar for KEDA.
Event Hub w/Kafka uses SASL_SSL authentication without certs with just PLAIN mechanism and needs specific TLS config while using the Sarama Go package.

TO DO:
- Update KEDA documentation to advertise this feature
- Add sample

### CODE REVIEW

**Comment by tomkerkhove** [comment]
Would you mind also sending a PR for the docs on https://github.com/kedacore/keda-docs

File: pkg/scalers/kafka_scaler.go [hunk 1]
```diff
@@ -49,6 +50,7 @@ const (
 	kafkaAuthModeForSaslScramSha256 kafkaAuthMode = "sasl_scram_sha256"
 	kafkaAuthModeForSaslScramSha512 kafkaAuthMode = "sasl_scram_sha512"
 	kafkaAuthModeForSaslSSL         kafkaAuthMode = "sasl_ssl"
+	kafkaAuthModeForSaslSSLPlain    kafkaAuthMode = "sasl_ssl_plain"
```
Comment (ahmelsayed) [inline pkg/scalers/kafka_scaler.go]: what's the difference between `sasl_ssl_plain` and `sasl_plaintext`, does it still use ssl but without a client cert?

File: pkg/scalers/kafka_scaler.go [hunk 2]
```diff
@@ -108,8 +110,10 @@ func parseKafkaMetadata(resolvedEnv, metadata, authParams map[string]string) (ka
 
 	meta.authMode = kafkaAuthModeForNone
 	if val, ok := authParams["authMode"]; ok {
+		val = strings.TrimSpace(val)
```
Comment (ahmelsayed) [inline pkg/scalers/kafka_scaler.go]: were you doing `echo 'value' | base64` by any chance?

File: pkg/scalers/kafka_scaler.go [hunk 3]
```diff
@@ -177,12 +181,25 @@ func getKafkaClients(metadata kafkaMetadata) (sarama.Client, sarama.ClusterAdmin
...
 		config.Net.SASL.Enable = true
 		config.Net.SASL.User = metadata.username
 		config.Net.SASL.Password = metadata.password
 	}
 
+	if metadata.authMode == kafkaAuthModeForSaslSSLPlain {
+		config.Net.SASL.Mechanism = sarama.SASLMechanism(sarama.SASLTypePlaintext)
+
+		tlsConfig := &tls.Config{
+			InsecureSkipVerify: true,
```
Comment (ahmelsayed) [inline pkg/scalers/kafka_scaler.go]: Is this needed? I'd imagine eventhubs must have a valid cert, no?

File: pkg/scalers/kafka_scaler.go (hunk 1 above)
Comment (rasavant-ms) [reply pkg/scalers/kafka_scaler.go]: That is correct, SASL_PLAINTEXT does not use SSL and the current SASL_SSL is configured to ask for certs.
Event Hubs authenticates with SASL_SSL with creds passed as PLAIN.

File: pkg/scalers/kafka_scaler.go (hunk 2 above)
Comment (rasavant-ms) [reply pkg/scalers/kafka_scaler.go]: Yes I am.. Is there another recommended way to do base64 encoding?

File: pkg/scalers/kafka_scaler.go (hunk 3 above)
Comment (rasavant-ms) [reply pkg/scalers/kafka_scaler.go]: This is the same configuration as recommended by the Event Hubs team in the sample GitHub repo here - https://github.com/Azure/azure-event-hubs-for-kafka/blob/master/quickstart/go-sarama-client/consumer/sarama-consumer.go#L89
The authentication is done using the connection string being passed in the `username` and `password` metadata

**Comment by rasavant-ms** [comment]
Closes #594

File: pkg/scalers/kafka_scaler.go (hunk 2 above)
Comment (ahmelsayed) [reply pkg/scalers/kafka_scaler.go]: `echo` just appends a new line character by default. Try `echo -n 'value' | base64` instead to tell it not to do that.

Comment (rasavant-ms) [reply pkg/scalers/kafka_scaler.go]: I will make sure to mention this in the documentation.

**Pull request merged**